src/main/java/com/feywild/feywild/entity/Shroomling.java

```diff
@@ -0,0 +1,279 @@
+                    if (Objects.equals(this.getCurrentPointOfInterest(), player.position())) {
+                        this.setCurrentTargetPos(this.blockPosition());
+                        player.sendMessage(new TranslatableComponent("message.feywild.shroomling_fey_stay").append(new TranslatableComponent("message.feywild.fey_stay").withStyle(ChatFormatting.ITALIC)), player.getUUID());
+                    } else {
+                        this.setCurrentTargetPos((BlockPos) null);
+                        player.sendMessage(new TranslatableComponent("message.feywild.shroomling_fey_follow").append(new TranslatableComponent("message.feywild.fey_follow").withStyle(ChatFormatting.ITALIC)), player.getUUID());
+                    }
+                }
+            }
+        }
+
+        ItemStack itemstack = player.getItemInHand(hand);
+        if (ComposterBlock.COMPOSTABLES.containsKey(itemstack.getItem()) &&
+                player == this.getOwner()) {
+            if (!level.isClientSide) {
+                if (!player.isCreative()) {
+                    player.getItemInHand(hand).shrink(1);
+                }
+                FeywildMod.getNetwork().sendParticles(this.level, ParticleSerializer.Type.CROPS_GROW, this.getX(), this.getY(), this.getZ());
+                player.swing(hand, true);
+                player.addItem(new ItemStack(Items.BONE_MEAL));
+            }
+            return InteractionResult.sidedSuccess(level.isClientSide);
+        }
+        return InteractionResult.CONSUME;
+    }
+
+    /* ANIMANTION */
+
+    @Override
+    public void registerControllers(AnimationData animationData) {
+        animationData.addAnimationController(new AnimationController<>(this, "controller", 0, this::animationPredicate));
+    }
+
+    private <E extends IAnimatable> PlayState animationPredicate(AnimationEvent<E> event) {
+        if (!this.dead && !this.isDeadOrDying()) {
+            if (this.getState() == State.WAVING) {
+                event.getController().setAnimation(new AnimationBuilder().addAnimation("animation.shroomling.wave", false));
+                return PlayState.CONTINUE;
+            } else if (this.getState() == State.SNEEZING) {
+                event.getController().setAnimation(new AnimationBuilder().addAnimation("animation.shroomling.sneeze", false));
+                return PlayState.CONTINUE;
+            }
+        }
+        if (this.getDeltaMovement().lengthSqr() < MIN_MOVING_SPEED_SQR) {
+            event.getController().setAnimation(new AnimationBuilder().addAnimation("animation.shroomling.static", true));
+        } else {
+            event.getController().setAnimation(new AnimationBuilder().addAnimation("animation.shroomling.walk", true));
+        }
+
+        return PlayState.CONTINUE;
+    }
+
+    public AnimationFactory getFactory() {
+        return this.factory;
+    }
+
+    @Override
+    public boolean isPersistenceRequired() {
+        return true; //unsure
```

Comment:
Make it a //TODO comment then, so we can easily see that we still need to address this.